app/controllers/form_controller.rb

```diff
@@ -50,7 +50,7 @@ def show_page
       page_id = request.path.split("/")[-1].underscore
       @page = @log.form.get_page(page_id)
       @subsection = @log.form.subsection_for_page(@page)
-      if @page.routed_to?(@log, current_user)
+      if @page.routed_to?(@log, current_user) && @log.collection_period_open?
```

Comment:
I suspect that you might want this for both `show_page` and `submit_form` — did you consider a `before_action` so stop execution and redirect if the collection period is closed?